route-rs-runtime/src/packet/ethernet.rs

```diff
@@ -0,0 +1,113 @@
+use std::borrow::Cow;
+use crate::packet::*;
+
+pub struct EthernetFrame<'packet> {
+    pub data: PacketData<'packet>,
+}
+
+impl<'frame> EthernetFrame<'frame> {
+    pub fn new(frame: PacketData) -> Result<EthernetFrame, &'static str> {
```

Comment:
Let's think about a legit `Error` type. I'm not too sure what the best way to do `Error`s is since `failure` became superseded by the stdlib though.

Reply:
Yeah, I would like to be able to return formatted strings or something.

Reply:
I'll take a look into Error types. Ideally we could return formatted strings